### ISSUE TITLE
Exception: java.lang.Boolean cannot be cast to io.swagger.v3.oas.models.media.Schema

### ISSUE THREAD

**Issue by datstarkey** [OpenAPITools/openapi-diff#256]
Hi,

Tool has been great so far, it's been working really well for our microservice architecture but one of our api's comes up with this error when trying to get the differences:

```
Unexpected exception. Reason: java.lang.Boolean cannot be cast to io.swagger.v3.oas.models.media.Schema
java.lang.ClassCastException: java.lang.Boolean cannot be cast to io.swagger.v3.oas.models.media.Schema
        at org.openapitools.openapidiff.core.compare.schemadiffresult.SchemaDiffResult.compareAdditionalProperties(SchemaDiffResult.java:133)
        at org.openapitools.openapidiff.core.compare.schemadiffresult.SchemaDiffResult.diff(SchemaDiffResult.java:78)
        at org.openapitools.openapidiff.core.compare.SchemaDiff.computeDiff(SchemaDiff.java:316)
        at org.openapitools.openapidiff.core.compare.SchemaDiff.computeDiff(SchemaDiff.java:27)
        at org.openapitools.openapidiff.core.compare.ReferenceDiffCache.cachedDiff(ReferenceDiffCache.java:44)
        at org.openapitools.openapidiff.core.compare.SchemaDiff.diff(SchemaDiff.java:282)
        at org.openapitools.openapidiff.core.compare.ContentDiff.diff(ContentDiff.java:33)
        at org.openapitools.openapidiff.core.compare.ResponseDiff.computeDiff(ResponseDiff.java:50)
        at org.openapitools.openapidiff.core.compare.ResponseDiff.computeDiff(ResponseDiff.java:15)
        at org.openapitools.openapidiff.core.compare.ReferenceDiffCache.cachedDiff(ReferenceDiffCache.java:51)
        at org.openapitools.openapidiff.core.compare.ResponseDiff.diff(ResponseDiff.java:34)
        at org.openapitools.openapidiff.core.compare.ApiResponseDiff.diff(ApiResponseDiff.java:32)
        at org.openapitools.openapidiff.core.compare.OperationDiff.diff(OperationDiff.java:63)
        at org.openapitools.openapidiff.core.compare.PathDiff.diff(PathDiff.java:35)
        at org.openapitools.openapidiff.core.compare.PathsDiff.lambda$diff$3(PathsDiff.java:83)
        at java.util.LinkedHashMap$LinkedKeySet.forEach(LinkedHashMap.java:559)
        at org.openapitools.openapidiff.core.compare.PathsDiff.diff(PathsDiff.java:41)
        at org.openapitools.openapidiff.core.compare.OpenApiDiff.compare(OpenApiDiff.java:95)
        at org.openapitools.openapidiff.core.compare.OpenApiDiff.compare(OpenApiDiff.java:66)
        at org.openapitools.openapidiff.core.OpenApiCompare.fromSpecifications(OpenApiCompare.java:101)
        at org.openapitools.openapidiff.core.OpenApiCompare.fromLocations(OpenApiCompare.java:90)
        at org.openapitools.openapidiff.core.OpenApiCompare.fromLocations(OpenApiCompare.java:77)
        at org.openapitools.openapidiff.cli.Main.main(Main.java:156)
```

Any help is apprecitated, many thanks.
Jake

**Comment by joschi** [comment]
@datstarkey Thanks for reporting!

Could you provide your OpenAPI spec(s) or a reduced version of the OpenAPI spec(s) to build a reproduction case? You can redact/anonymize the contents of the OpenAPI specs as long as they trigger the bug you've reported.

**Comment by datstarkey** [comment]
[newVersionJson.txt](https://github.com/OpenAPITools/openapi-diff/files/7002930/newVersionJson.txt)
[oldVersionjson.txt](https://github.com/OpenAPITools/openapi-diff/files/7002931/oldVersionjson.txt)

I've attached them both json objects as text files here, comparing these two is giving me the error.

Thanks

**Comment by joschi** [comment]
@datstarkey Thanks for reporting this! The issue will be fixed in the next beta of OpenAPI-diff 2.0.0.
